sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -237,53 +237,58 @@ case class In(value: Expression, list: Seq[Expression]) extends Predicate {
     val javaDataType = ctx.javaType(value.dataType)
     val valueGen = value.genCode(ctx)
     val listGen = list.map(_.genCode(ctx))
-    ctx.addMutableState(ctx.JAVA_BOOLEAN, ev.value)
-    ctx.addMutableState(ctx.JAVA_BOOLEAN, ev.isNull)
+    // inValue  -1:isNull, 0:false, 1:true
+    val inValue = ctx.freshName("value")
```

Comment:
nit: `inTmpResult`?

Reply:
`// inValue  -1 indicates at lease one expr in list is evaluated to null. 0 means no matches found. 1 means the expr in list matches the given value expr.`